### PR TITLE
🐛 Source MySQl: Fixed MySQL CDC Decimal handling

### LINKED ISSUE #4105: MySQL CDC Decimal handling
## Expected Behavior
MySQL Decimal values should be converted to a numeric value in the raw data and then converted to a float in the normalization step.

## Current Behavior
I currently have a MySQL table (connected to BigQuery) with column of type decimal(10,4) where this decimal value is converted to a string. For example, the value 0.188 is awlays mapped to the string "B1g=" and 0.311 is mapped to "DCY=".
The subsequent normalization step then fails since the genreated SQL attempts to cast the string to a float.

## Logs

[logs-37-0.txt](https://github.com/airbytehq/airbyte/files/6650122/logs-37-0.txt)

## Steps to Reproduce
1. Create and fill a MySQL table with decimal columns
2. Set up Airbyte MySQL source conenctor with CDC
3. Set up connection to BigQuery
4. Run sync

## Severity of the bug for you
High

## Airbyte Version
0.24.7-alpha

## Connector Version (if applicable)
0.3.7

### PR DESCRIPTION
**Pull request by etsybaev**
## What
Currently a MySQL table with column of type decimal(10,4) where this decimal value is converted to a string. For example, the value 0.188 is always mapped to the string "B1g=" and 0.311 is mapped to "DCY=".

## How
Updated Debezium to store decimal values as a String representation in its own Kafka messages instead of binary.

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [x] PR merged into master branch
- [ ] Follow up tickets have been created
- [x] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by yaroslav-hrytsaienko — APPROVED** [review]

**Comment by subodh1810** [comment]
According to the docs https://debezium.io/documentation/reference/1.4/connectors/mysql.html#mysql-property-decimal-handling-mode
we were using the `precise` mode by default but now switching to `double`. My concern is that if we make the change, it would result in a loss of precision. Also can you explain how exactly we were ending up with string and not original values? 

```
Specifies how the connector should handle values for DECIMAL and NUMERIC columns:

precise (the default) represents them precisely using java.math.BigDecimal values represented in change events in a binary form.

double represents them using double values, which may result in a loss of precision but is easier to use.

string encodes values as formatted strings, which is easy to consume but semantic information about the real type is lost.
```

**Comment by etsybaev** [comment]
Hi @subodh1810. Unfortunately, I do not see a reply button under your comment, so will add it as another comment instead of a thread.  This happens when debezium reads values from DB and saves them on its kafka as far as I understood.  In the original ticket, there was a comment on how to fix it so I followed this recommendation. Checked using test and it seems to solve the issue.

**Comment by etsybaev** [comment]
Hi @subodh1810.

Checked it one more time. It seems like for now in master branch it's set by default as "decimal.handing.mode=precise" which's caused returning this value as a binary. But there is another option instead of "double". We may set it as a String explicitly, we will still parse it as a Decimal SQL type, but if SQL type is set to have 5 digits after comma (decimal(10,5)) - then the value like "0.188" would be returned with 0 at the end (ex. "0.18800"). But in this case of transferring it via debezium's kafka as a String - we wouldn't have an issue with a loss of precision. Shall I change it to String, what do you think?

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/DebeziumRecordPublisher.java [hunk 1]
```diff
@@ -152,6 +152,7 @@ protected Properties getDebeziumProperties(JsonNode config,
     // https://debezium.io/documentation/reference/connectors/mysql.html#mysql-boolean-values
     props.setProperty("converters", "boolean");
     props.setProperty("boolean.type", "io.debezium.connector.mysql.converters.TinyIntOneToBooleanConverter");
+    props.setProperty("decimal.handling.mode", "string");
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/DebeziumRecordPublisher.java:155]: can you add a comment explaining: 
1. why we set this property 
2. whether it results in a loss of precision

Comment (etsybaev) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/DebeziumRecordPublisher.java:162]: Done, thanks

**Review by sherifnada — APPROVED** [review]

**Pull request merged**